File: Backend/TaskManagement.API/Models/Category.cs
namespace TaskManagement.API.Models;

public class Category
{
    public Guid Id {get; set;}
    public string Name {get; set;} = string.Empty;
    public string? Description {get; set;}
    public string Color {get; set;} = "#007bff";
    public Guid UserId {get; set;}
    public DateTime CreatedAt {get; set;} = DateTime.UtcNow;

    // Navigation Property'ler
    public User User {get; set;} = null!;
    public ICollection<TaskItem> Tasks {get; set;} = new List<TaskItem>();
}

File: Frontend/TaskManagement.Web/src/types/category.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
  color: string;
  userId: string;
  createdAt: string;
}

export interface CreateCategory {
  name: string;
  description?: string;
  color?: string;
}

export interface UpdateCategory {
  name?: string;
  description?: string;
  color?: string;
}
